Logger/LogFactory.cs

```diff
@@ -2,9 +2,25 @@
 
 public class LogFactory
 {
-    public BaseLogger CreateLogger(string className)
-    {
+    private string? _loggerPath;
 
-        return null;
+    public BaseLogger? CreateLogger (string className)
+    {
+        if (className == nameof(FileLogger))
+        {
+            FileLogger FileLogger = new(_loggerPath!) { ClassName = className };
+            return FileLogger;
+        } else
+        {
+            return null;
+        }
```

Comment:
Definitely not wrong but you could consider removing the else and just have return null by itself,  if the if statement didn't trigger then we wouldn't get that earlier return FileLogger and would instead be executing whatever is after the if statement anyway.
```suggestion
        }
        
        return null;
        
```

Reply:
Ahh thanks for the catch and explanation will fix